### PR TITLE
GH Actions

### PR DESCRIPTION
**Pull request by ssiccha**
Based on https://github.com/gap-system/gap/pull/4036. See https://github.com/ssiccha/recog/actions for the runs.

### CODE REVIEW

**Comment by ssiccha** [comment]
I think I'm done. Feedback welcome. :)

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        test-suites: ["testinstall"]
+        branch: ["master", "stable-4.11", "stable-4.10"]
+
+    env:
+      TEST_SUITES: ${{ matrix.test-suites }}
+      GAP_PKGS_TO_CLONE: "AutoDoc"
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: "Set up Python 3.7"
```
Comment (fingolfin) [inline .github/workflows/CI.yml:25]: What is this needed for?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      - name: "Download CI scripts"
+        run:  git clone https://github.com/gap-system/pkg-ci-scripts.git scripts
+      - name: "Compile GAP and download packages"
+        env:
+          GAPBRANCH: ${{ matrix.branch }}
+        run:  bash scripts/build_gap.sh
+      - name: "Prepare this package"
+        run:  bash scripts/build_pkg.sh
+      - name: "Compile documentation"
+        run:  echo "QUIT;" | $HOME/gap/bin/gap.sh -l "$PWD/gaproot;" --quitonbreak makedoc.g
```
Comment (fingolfin) [inline .github/workflows/CI.yml:39]:
```suggestion
        run:  $HOME/gap/bin/gap.sh -l "$PWD/gaproot;" --quitonbreak makedoc.g -c "QUIT;"
```

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  manual:
+    name: Build manuals
+    runs-on: ubuntu-latest
+
+    env:
+      GAP_PKGS_TO_CLONE: "AutoDoc"
+    steps:
+      - uses: actions/checkout@v2
+      - name: "Set up Python 3.7"
```
Comment (fingolfin) [inline .github/workflows/CI.yml:55]: Why is this needed?

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        test-suites: ["testinstall"]
+        branch: ["master", "stable-4.11", "stable-4.10"]
+
+    env:
+      TEST_SUITES: ${{ matrix.test-suites }}
```
Comment (fingolfin) [inline .github/workflows/CI.yml:20]:
```suggestion
```
This has no meaning (if at all, one would override `GAP_TESTFILE`)

File: .github/workflows/CI.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
+name: CI
+
+# Trigger the workflow on push or pull request
+on:
+  - push
+  - pull_request
+
+jobs:
+  test:
+    name: ${{ matrix.test-suites }} - ${{ matrix.branch }} - ${{ matrix.os }}
```
Comment (fingolfin) [inline .github/workflows/CI.yml:10]:
```suggestion
    name: ${{ matrix.branch }} - ${{ matrix.os }}
```

File: .github/workflows/CI.yml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            texlive-extra-utils
+            texlive-fonts-recommended texlive-fonts-extra
+          )
+          sudo apt-get install "${packages[@]}"
+      - name: "Compile GAP and download packages"
+        run:  bash scripts/build_gap.sh
+      - name: "Prepare this package"
+        run:  bash scripts/build_pkg.sh
+      - name: "Compile documentation"
+        run:  echo "QUIT;" | $HOME/gap/bin/gap.sh -l "$PWD/gaproot;" --quitonbreak makedoc.g
```
Comment (fingolfin) [inline .github/workflows/CI.yml:76]: See above

File: .github/workflows/CI.yml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      - name: "Upload documentation"
+        uses: actions/upload-artifact@v1
```
Comment (fingolfin) [inline .github/workflows/CI.yml:78]: Nice!

File: .github/workflows/CI.yml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+jobs:
+  test:
+    name: ${{ matrix.test-suites }} - ${{ matrix.branch }} - ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        test-suites: ["testinstall"]
```
Comment (fingolfin) [inline .github/workflows/CI.yml:16]:
```suggestion
```

File: .github/workflows/CI.yml [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        branch: ["master", "stable-4.11", "stable-4.10"]
```
Comment (fingolfin) [inline .github/workflows/CI.yml:17]: I would suggest to put this onto multiple lines, as in the old Travis setup, so that adding new GAP versions has a minimal diff
```suggestion
        branch:
          - master
          - stable-4.11
          - stable-4.10
```

**Review by fingolfin — COMMENTED** [review]
Great! I have some minor remarks.

Can you also update the CI badge and link in README.md, and remove `.travis.yml`?

**Comment by fingolfin** [comment]
On the long run, we could also investigate using Docker containers with prebuilt GAP. See https://github.com/homalg-project/CatReps/blob/master/.github/workflows/Tests.yml and https://github.com/homalg-project/homalg_project/blob/master/.github/workflows/Tests.yml

File: .github/workflows/CI.yml (hunk 6 above)
Comment (fingolfin) [inline .github/workflows/CI.yml:76]: Maybe also set the environment variable `SOURCE_DATE_EPOCH` to avoid timestamps in manual PDFS, which should ensure reproducible PDFs (i.e. building it twice should give identical binaries)

Comment (ssiccha) [reply .github/workflows/CI.yml:76]: That didn't work on my machine. Is there maybe a typo somewhere? I grepped for `SOURCE_DATE_EPOCH`, `DATE` and `EPOCH` in gap and in GAPDoc but didn't find anything.

Comment (ssiccha) [reply .github/workflows/CI.yml]: I set up a separate issues for this.

File: .github/workflows/CI.yml [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+      matrix:
+        os: [ubuntu-latest]
+        branch:
+          - master
+          - stable-4.11
+          - stable-4.10
+
+    env:
+      GAP_PKGS_TO_CLONE: "AutoDoc"
+      SOURCE_DATE_EPOCH: "AutoDoc"
```
Comment (fingolfin) [inline .github/workflows/CI.yml:23]:
```suggestion
      SOURCE_DATE_EPOCH: 0
```

**Review by fingolfin — APPROVED** [review]
